.github/workflows/npm-publish.yml

```diff
@@ -0,0 +1,33 @@
+name: Build and publish NPM package
+
+on:
+  workflow_dispatch:
+    inputs:
+      node_version:
+        default: '19'
+        description: 'Node version to use'
+        required: true
+        type: string
+
+jobs:
+  build-package:
+    environment: release
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3.3.0
+        with:
+          fetch-depth: 0 # npm run prod requires tags, sadly this is the only way to get them.
+      - uses: actions/setup-node@v3
+        with:
+          node-version: ${{ inputs.node_version }}
+      - name: Install dependencies
+        run: npm install
+      - name: Test with npm
+        run: npm run test
```

Comment:
I found it rather unexpected that running tests in the workflow didn't require installing dev dependencies.

Jest, the linters, and webpack should all fall under devDependencies. However, I would also propose that since we're publishing the built and minified JS file to NPM, all of our dependencies should be classified as devDependencies because most users who install the package won't require any of them. Additionally, I believe it's time we committed the package lock file. @paulwarren-wk @aaroncameron-wk what are your thoughts?

Reply:
Good catch 👍

I think all of the build and test dependencies can still go under dev, but yes, leave the imported dependencies in prod.